web/pages/settings/credits.tsx

```diff
@@ -7,140 +7,287 @@ import { Switch } from "@/components/ui/switch";
 import { Badge } from "@/components/ui/badge";
 import { Small, XSmall, Muted } from "@/components/ui/typography";
 import {
-  RefreshCcw,
-  Settings,
-  ExternalLink,
-  FileText,
-  ChevronLeft,
-  ChevronRight,
-} from "lucide-react";
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { RefreshCcw, ChevronLeft, ChevronRight } from "lucide-react";
+import {
+  useCredits,
+  useCreditTransactions,
+  useCreateCheckoutSession,
+  type PurchasedCredits,
+} from "@/services/hooks/useCredits";
+import Link from "next/link";
+import { formatDate } from "@/utils/date";
+import PaymentModal from "@/components/templates/settings/PaymentModal";
 
 const CreditsSettings: NextPageWithLayout<void> = () => {
   const [autoTopUpEnabled, setAutoTopUpEnabled] = useState(false);
-  const [currentPage, setCurrentPage] = useState(1);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [pageSize, setPageSize] = useState(5);
+  const [isPaymentModalOpen, setIsPaymentModalOpen] = useState(false);
+
+  const {
+    data: creditData,
+    isLoading,
+    error: creditError,
+    refetch,
+  } = useCredits();
+  const {
+    data: transactionsData,
+    isLoading: transactionsLoading,
+    refetch: refetchTransactions,
+    error: transactionsError,
+  } = useCreditTransactions({
+    page: currentPage,
+    pageSize: pageSize,
+  });
+
+  const transactions = transactionsData?.purchases || [];
+  const totalTransactions = transactionsData?.total || 0;
+  const totalPages = Math.ceil(totalTransactions / pageSize);
+  const hasMore = currentPage < totalPages - 1;
+  const hasPrevious = currentPage > 0;
 
-  // Mock data - replace with actual data fetching
-  const currentBalance = 0;
-  const transactions: any[] = [];
+  const createCheckoutSession = useCreateCheckoutSession();
+  const handlePaymentSubmit = (amount: number) => {
+    createCheckoutSession.mutate(amount);
+  };
```

Comment:
nit: this is redunant imo, you can just pass createCheckoutSession.mutate to the onclick handler